src/cockpit/channels/filesystem.py

```diff
@@ -266,7 +275,7 @@ def chown_if_required(fd: 'int', buf: 'os.stat_result | None' = None):
                     if tag != tag_from_stat(buf):
                         raise ChannelError('change-conflict')
                     # chown/chmod from the existing file permissions
-                    os.fchmod(fd, stat.S_IMODE(buf.st_mode))
+                    apply_file_mode(fd, stat.S_IMODE(buf.st_mode))
```

Comment:
...and now I'm fairly sure this is wrong.  In this case if a tag and attrs are specified then the mode from the old file is going to win, but the owner/group from the attrs will win.